tests/test_reference.py

```diff
@@ -0,0 +1,492 @@
+            [
+                ReferencePathElement(
+                    type_=ReferencePathElementType.ACTIVE,
+                    source="class_a",
+                    slot="has_class_b",
+                    target="class_b",
+                ),
+                ReferencePathElement(
+                    type_=ReferencePathElementType.PASSIVE,
+                    source="class_b",
+                    slot="has_class_b",
+                    target="class_c",
+                ),
+            ],
+        ),
+    ],
+)
+def test_path_str_to_object_elements(
+    path_str: str, expected_elements: ReferencePathElement
+):
+    """Test the path_str_to_object_elements method."""
+
+    observed_elements = path_str_to_object_elements(path_str)
+    assert observed_elements == expected_elements
+
+
+@pytest.mark.parametrize(
+    "path_str, expected_elements, expected_source, expected_target",
+    [
+        (
+            "class_a(has_class_b)>class_b",
+            [
+                ReferencePathElement(
+                    type_=ReferencePathElementType.ACTIVE,
+                    source="class_a",
+                    slot="has_class_b",
+                    target="class_b",
+                )
+            ],
+            "class_a",
+            "class_b",
+        ),
+        (
+            "class_a<(has_class_a)class_b",
+            [
+                ReferencePathElement(
+                    type_=ReferencePathElementType.PASSIVE,
+                    source="class_a",
+                    slot="has_class_a",
+                    target="class_b",
+                )
+            ],
+            "class_a",
+            "class_b",
+        ),
+        (
+            "class_a(has_class_b)>class_b(has_class_c)>class_c",
+            [
+                ReferencePathElement(
+                    type_=ReferencePathElementType.ACTIVE,
+                    source="class_a",
+                    slot="has_class_b",
+                    target="class_b",
+                ),
+                ReferencePathElement(
+                    type_=ReferencePathElementType.ACTIVE,
+                    source="class_b",
+                    slot="has_class_c",
+                    target="class_c",
+                ),
+            ],
+            "class_a",
+            "class_c",
+        ),
+        (
+            "class_a(has_class_b)>class_b<(has_class_b)class_c",
+            [
+                ReferencePathElement(
+                    type_=ReferencePathElementType.ACTIVE,
+                    source="class_a",
+                    slot="has_class_b",
+                    target="class_b",
+                ),
+                ReferencePathElement(
+                    type_=ReferencePathElementType.PASSIVE,
+                    source="class_b",
+                    slot="has_class_b",
+                    target="class_c",
+                ),
+            ],
+            "class_a",
+            "class_c",
+        ),
+    ],
+)
```

Comment:
Maybe also add an example where the path contains whitespace to test its automatic removal.

Reply:
Good idea. Added that.